### DISCUSSION TITLE
close a conversation with a message

### DISCUSSION THREAD

**Discussion by mjavadtatari** [kianahmadian/python-bale-bot#79] [Q&A] [answered]
سلام وقت بخیر
یه سوال داشتم در مورد conversation ها
دو حالت رو داخل مثال قرار دادین که یا timeout وجود داره یا خیر.
برای اون مدلی که timeout ست نشده، آیا راهی وجود داره که کاربر بتونه با وارد کردن دستوری مثل `/cancel` حالت client.wait_for رو close کنه؟

**Comment by kian-ahmadian** [comment, marked as answer]
سلام عزیز وقتتون بخیر
این قابلیت که در هر کجای ربات که باشید امکان حذف حالت wait_for وجود داشته باشه در نسخه `2.5.1` منتشر میشه

> در نظر داشته باشید در نسخه بعدی بخش های مربوط به انتظار برای دریافت یک پیام و یا ایونت جدید دچار تغییر خواهند شد؛ بنابراین پیشنهاد میشه کانال های اطلاع رسانی و یا گیت هاب پروژه رو تحت نظر داشته باشید

اما در حال حاضر میتونید ایونت مورد نظر خودتون در متد `client.wait_for` رو به `on_update` تغییر بدید و علاوه بر بررسی هایی که احتیاج هست نسبت به عملکرد مورد نظر خودتون داشته باشید، بررسی کنید که آپدیت ارسال شده پیام هست و اون پیام محتواش شامل پیغام لغو میشه یا خیر

## مثالی از این عملکرد (انتظار برای دریافت یک پیام):
```python
...
await message.reply('what is your name?')
def answer_checker(update: bale.Update):
        return (
            update.message and
            message.author == update.message.author and message.chat == update.message.chat and bool(update.message.content)
)

answer_obj: bale.Update = await client.wait_for('update', check=answer_checker)
if not answer_obj.message.content == '/cancel':
     return await answer_obj.reply(f'Your name is {answer_obj.content}')
``` 

اگر مشکلی بود در خدمتم

**Reply by kian-ahmadian** [reply]
**اما همونطور که در پیام قبل هم به عرض رسانده شد، در نسخه بعدی هندلر ها می آیند و بخش ایونت ها حذف خواهد شد**
